app/forms/post_form.rb

```diff
@@ -0,0 +1,56 @@
+class PostForm
+  include ActiveModel::Model
+  include Virtus.model
+  extend CarrierWave::Mount
+
+  validates :title,   presence: true, length: { maximum: 50 }
+  validates :content, presence: true, length: { maximum: 10000 }
+
+  attribute :user_id,     Integer
+  attribute :category_id, Integer
+  attribute :title,       String
+  attribute :content,     String
+  attribute :status,      Boolean
+
+  mount_uploader :picture, PictureUploader
+
+  attr_accessor :pictures
+
+  def initialize(post = Post.new)
+    @post = post
+    self.attributes = @post.attributes if @post.persisted?
+  end
+
+  def assign_attributes(params = {})
+    @params = params
+    # 画像情報からインスタンスを生成する
+    attributes = params.to_h[:pictures_attributes]
+    @pictures ||= []
+    attributes&.map do |attribute|
+      picture = Picture.new(attribute)
+      @pictures.push(picture)
+    end
```

Comment:
気になったのが、attributesにすると、self.attributesと同義になるので、self.attributesに格納してるんじゃないかなと思いました。
どちらにせよ可読性的に紛らわしいので変数名を変えた方がいいかと思います。
あと、to_hしなくても取れませんかね？
```suggestion
    pictures_attributes = @params[:pictures_attributes]
    @pictures ||= []
    pictures_attributes&.map do |pictures_attribute|
      picture = Picture.new(pictures_attribute)
      @pictures.push(picture)
    end
```

Reply:
一応attributesでも動いていましたが、誤解を生む記述だったので
指摘いただいた通りに変更しました。ありがとうございます。